XMonad/Util/Hacks.hs

```diff
@@ -92,3 +99,47 @@ javaHack conf = conf
   { startupHook = startupHook conf
                     *> io (putEnv "_JAVA_AWT_WM_NONREPARENTING=1")
   }
+
+
+-- $raiseTrayer
+-- Placing @trayer@ on top of @xmobar@ is somewhat tricky:
+--
+-- - they both should be lowered to the bottom of the stacking order to avoid
+--   overlapping fullscreen windows
+--
+-- - the tray needs to be stacked on top of the panel regardless of which
+--   happens to start first
+--
+-- 'trayerAboveXmobarEventHook' (and the more generic
+-- 'trayAbovePanelEventHook') is an event hook that ensures the latter:
+-- whenever the tray lowers itself to the bottom of the stack, it checks
+-- whether there are any panels above it and lowers these again.
+--
+-- To ensure the former, that is having both @trayer@ and @xmobar@ lower
+-- themselves, which is a necessary prerequisite for this event hook to
+-- trigger:
+--
+-- - set @lowerOnStart = True@ and @overrideRedirect = True@ in @~/.xmobarrc@
+-- - pass @-l@ to @trayer@
+
+-- | 'trayAbovePanelEventHook' for trayer/xmobar
+trayerAboveXmobarEventHook :: Event -> X All
+trayerAboveXmobarEventHook = trayAbovePanelEventHook (className =? "trayer") (appName =? "xmobar")
+
+-- | Whenever a tray window lowers itself to the bottom of the stack, look for
+-- any panels above it and lower these.
+trayAbovePanelEventHook
+  :: Query Bool -- ^ tray
+  -> Query Bool -- ^ panel
+  -> (Event -> X All) -- ^ event hook
+trayAbovePanelEventHook trayQ panelQ ConfigureEvent{ev_window = w, ev_above = a} | a == none = do
+  -- when tray is lowered to the bottom of stack (a == none),
+  -- put all panels that are above it below
```

Comment:
Maybe "move all panels below it"?

Reply:
Hm, do I need this comment at all? It's saying the same thing as the haddock for the function itself (which I wrote today and then forgot to take another look at the comment in code).

Reply:
I dropped it. And then I added a usage to the the doc which I forgot.